fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/transactionprocessor/AbstractLoanRepaymentScheduleTransactionProcessor.java

```diff
@@ -395,6 +397,16 @@ protected void reprocessInstallments(List<LoanRepaymentScheduleInstallment> inst
             if (loanRepaymentScheduleInstallment.getTotalOutstanding(currency).isGreaterThanZero()) {
                 loanRepaymentScheduleInstallment.updateObligationMet(false);
                 loanRepaymentScheduleInstallment.updateObligationMetOnDate(null);
+            } else {
```

Comment:
please check whether the obligation met is `false` and there is no `obligationsMetDate` already set. We need to calculate only if thats the case

Reply:
done.